theseus/src/util/export.rs

```diff
@@ -0,0 +1,298 @@
+
+    // Initialize loading bar
+    let loading_bar = if loading_bar {
+        Some(
+            init_loading(
+                LoadingBarType::ZipExtract {
+                    profile_path: profile.path.to_path_buf(),
+                    profile_name: profile.metadata.name.clone(),
+                },
+                path_list.len() as f64,
+                "Exporting profile to .mrpack",
+            )
+            .await?,
+        )
+    } else {
+        None
+    };
+
+    // Iterate over every file in the folder
+    // Every file that is NOT in the config file is added to the zip, in overrides
+    for path in path_list {
+        if let Some(ref loading_bar) = loading_bar {
+            emit_loading(loading_bar, 1.0, None).await?;
+        }
+
+        // Get local path of file, relative to profile folder
+        let name = path.strip_prefix(profile_base_path)?;
+
+        // Get highest level folder pair ('a/b' in 'a/b/c', 'a' in 'a')
+        // We only go one layer deep for the sake of not having a huge list of overrides
+        let topmost_two = name
+            .iter()
+            .take(2)
+            .map(|os| os.to_string_lossy().to_string())
+            .collect::<Vec<_>>();
+
+        // a,b => a/b
+        // a => a
+        let topmost = match topmost_two.len() {
+            2 => topmost_two.join("/"),
+            1 => topmost_two[0].clone(),
+            _ => {
+                return Err(crate::ErrorKind::OtherError(
+                    "No topmost folder found".to_string(),
+                )
+                .into())
+            }
+        };
+
+        if !included_overrides.contains(&topmost) {
+            continue;
+        }
+
+        let name: std::borrow::Cow<str> = name.to_string_lossy();
+        let name = name.replace('\\', "/");
+        let name = name.trim_start_matches('/').to_string();
+
+        if modrinth_path_list.contains(&name) {
+            continue;
+        }
+
+        // File is not in the config file, add it to the .mrpack zip
+        if path.is_file() {
+            let mut file = File::open(&path).await?;
+            let mut data = Vec::new();
+            file.read_to_end(&mut data).await?;
+            let builder = ZipEntryBuilder::new(
+                format!("overrides/{name}"),
+                Compression::Deflate,
+            );
+            writer.write_entry_whole(builder, &data).await?;
+        }
+    }
+
+    // Add modrinth json to the zip
+    let data = serde_json::to_vec_pretty(&packfile)?;
+    let builder = ZipEntryBuilder::new(
+        "modrinth.index.json".to_string(),
+        Compression::Deflate,
+    );
+    writer.write_entry_whole(builder, &data).await?;
+
+    writer.close().await?;
+    Ok(())
+}
+
+/// Creates a json configuration for a .mrpack zipped file
+// Version ID of uploaded version (ie 1.1.5), not the unique identifying ID of the version (nvrqJg44)
+#[tracing::instrument]
+pub fn create_mrpack_json(
```

Comment:
Profile project files which aren't modrinth projects should be exported in overrides (unless the user filters the files out). This should also check included files to see if the path is included.